controllers/topicsController.js

```diff
@@ -21,29 +22,46 @@ exports.getApis = (req, res, next) => {
 
 exports.getArticle = (req, res, next) => {
   const id = req.params.article_id;
-  getArticleById(id).then((article) => {
-    res.status(200).send({ article });
-  })
-  .catch(next)
-}
+  getArticleById(id)
+    .then((article) => {
+      res.status(200).send({ article });
+    })
+    .catch(next);
+};
+
+exports.getArticles = (req, res, next) => {
+  getAllArticles().then((data) => {
+    res.status(200).send({ articles: data });
+  });
+};
 
-exports.getArticles= (req, res, next) => {
-    getAllArticles()
-      .then((data) => {
-        res.status(200).send({ articles: data });
-      })
-  };
+exports.getComments = (req, res, next) => {
+  const { article_id } = req.params;
+  const commentsPromises = [getCommentsByArticleId(article_id)];
+  if (article_id) {
+    commentsPromises.push(getArticleById(article_id));
+  }
 
-exports.getComments = (req, res, next) =>{
-    const {article_id} = req.params
-    const commentsPromises = [getCommentsByArticleId(article_id)]
-    if(article_id) {
-        commentsPromises.push(getArticleById(article_id))
-    }
+  Promise.all(commentsPromises)
+    .then((resolvedPromises) => {
+      const comments = resolvedPromises[0];
+      res.status(200).send({ comments });
+    })
+    .catch(next);
+};
 
-    Promise.all(commentsPromises).then((resolvedPromises)=>{
-        const comments = resolvedPromises[0]
-        res.status(200).send({comments})
+exports.postComment = (req, res, next) => {
+  const newComment = req.body;
+  const { article_id } = req.params;
+  
+  const commentPromises = [
+      getArticleById(article_id),
+      getUserByUsername(newComment.username),
+      postCommentToArticle(newComment, article_id),
+  ]
```

Comment:
Nice use of a Promise.all to invoke several models to check for the existence of your user and article.

You may find that PSQL handles some of these checks for you because of foreign key constraints, as a comment cannot be inserted with a username that doesn't exist. However by using your Promise.all you get a lot of flexibility on your error messages. Pros and cons to both!